pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,136 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	BareMetal *BareMetal `json:"bareMetal"` // mandatory for now, possibly optional in future
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	UserManagedNetworking  bool   `json:"userManagedNetworking,omitempty"`
+	AdditionalNtpSource    string `json:"additionalNtpSource,omitempty"`
+	InstallConfigOverrides string `json:"installConfigOverrides,omitempty"`
+
+	SshPublicKey string `json:"sshPublicKey,omitempty"`
+}
+
+// BareMetal contains agent based install configuration specific to bare metal clusters.
+type BareMetal struct {
+	APIVIP            string `json:"apiVIP"`
+	APIVIPDNSName     string `json:"apiVIPDNSName"`
+	IngressVIP        string `json:"ingressVIP"`
+	VIPDHCPAllocation bool   `json:"vipDHCPAllocation"`
```

Comment:
`VIPDHCPAllocation` what does this field do? auto assign all the VIPS requested above?

i would recommend using union discriminator like https://github.com/openshift/api/blob/790e0a84e3e01dc3f6218e655b0e92b809ae7fd0/config/v1/types_infrastructure.go#L145-L146 here.

enumA -> auto assign
enumB -> user provided.

Reply:
Yes assigns VIPs using DHCP